fdbserver/fdbserver.actor.cpp

```diff
@@ -1331,10 +1331,14 @@ int main(int argc, char* argv[]) {
 			flushAndExit(FDB_EXIT_ERROR);
 		}
 
-		if( role == FDBD && publicAddresses.empty()) {
-			fprintf(stderr, "ERROR: The -p or --public_address option is required\n");
-			printHelpTeaser(argv[0]);
-			flushAndExit(FDB_EXIT_ERROR);
+		if (publicAddresses.empty()) {
+			if (role == FDBD || role == NetworkTestServer || role == Restore) {
```

Comment:
I'm assuming that this is that this check needs to be the same as the one at line 1489.

Can we just move the printing of the error to that `if` block?  It looks like the only other thing using publicAddresses is the `openTraceFile()` call on line 1471, which you could just pass a NetworkAddress() to if empty?

Or is there no real chance of this changing in the future, and thus it doesn't really matter what we do?

Reply:
Yes, you are right. Should avoid this duplicate expression. I've moved the code and avoiding that duplicate expression inside if, but I'm checking for error just before opening trace file so that we can avoid creating trace files unnecessarily (which isn't really a concern except for chronic `ls -al` bashers like me).